dts/bindings/audio/st,mp34dt05.yaml

```diff
@@ -0,0 +1,25 @@
+#
+# Copyright (c) 2018 STMicroelectronics
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+---
+title: ST Microelectronics MP34DT05 pdm microphone
+version: 0.1
+
```

Comment:
Add here:
```
parent:
    bus: i2s
```
Then, in i2s.yaml, add the following:
```
child:
    bus: i2s
```
Then, you'll get a DT_MP34DT05_MASTER_DEV_NAME that you'll be able to use in driver when binding driver bus controller

Reply:
After doing this above I'm able to obtain:
#define DT_ST_STM32_I2S_40015000_ST_MP34DT05_40015000_MP34DT05_BUS_NAME            "I2S_5"

Very long BUS_NAME. 
But at the end I should be able to do, in dts_fixup,

#define DT_MP34DT05_MASTER_DEV_NAME 
 DT_ST_STM32_I2S_40015000_ST_MP34DT05_40015000_MP34DT05_BUS_NAME

Reply:
What do you think?
I was expecting a shorter name, like

DT_ST_STM32_I2S_40015000_ST_MP34DT05_BUS_NAME

Reply:
@erwango
CHanged and committed.
Now this define will be generated:

#define DT_ST_STM32_I2S_40015000_ST_MP34DT05_40015000_MP34DT05_BUS_NAME "I2S_5"

It is used to define DT_MP34DT05_MASTER_DEV_NAME which will be used in mp34dt05 driver.

Reply:
Indeed, we should anyway a name that has no address reference, so we'll need a fixup indeed.
Then, there might be an issue in generation script and 'ST_MP34DT05_40015000_MP34DT05_BUS_NAME' looks indeed strange. Maybe this is coming from the fact there is no reg associated to the mic node. I'll check tomorrow.

Meanwhile, can you push using the fixup you proposed?

Reply:
Yes, I already did it yesterday.
I followed your instructions and added

 parent:
      bus: i2s

in microphone yaml file.

Reply:
So I confirm that string 'DT_ST_STM32_I2S_40015000_ST_MP34DT05_40015000_MP34DT05_BUS_NAME' is due to lack of reg in 96b_argonkey mp34dt05 node.
With a reg we get something like DT_ST_STM32_I2S_40015000_ST_MP34DT05_1_BUS_NAME.

If no reg is as expected we need to hack extract_dts_include script to take this into account
Do you confirm this is as expected. Isn't it possible to have several mic on the same bus?

Reply:
For 'reg' you mean 'address', like hts221@5f for I2C case?
I'm not a big expert of I2S bus, but as far as I understood there is no 'address' concept in I2S. It is possible indeed to have two microphones attached to same I2S controller (i.e. for audio stereo), but they are seen as one from controller point of view. It uses I2S_WS wire to sample mic1 (WS low) and mic2 (WS hi) alternatevely.

Nevertheless, in this case (stereo with 2 mics) I guess that we need to declare both as subnodes of the same I2S controller, so we probably need to use '0' and '1' like mp34dt05@0 and mp34dt05@1.

Correct me if I'm wrong, but I think that this goes back to #9034 where we had a discussion with address-cell and size.